commands/azureCommands/pull-from-azure.ts

```diff
@@ -0,0 +1,38 @@
+import vscode = require('vscode');
+import { reporter } from '../../telemetry/telemetry';
+const teleCmdId: string = 'vscode-docker.image.pullFromAzure';
+import { AzureImageNode } from '../../explorer/models/AzureRegistryNodes';
+import * as acrTools from '../../utils/Azure/acrTools';
+
+/* Pulls an image from Azure. The context is the image node the user has right clicked on */
+export async function pullFromAzure(context?: AzureImageNode): Promise<any> {
+
+    // Step 1: Using loginCredentials() function to get the username and password. This takes care of all users, even if they don't have the Azure CLI
+    let credentials;
+    try {
+        credentials = await acrTools.loginCredentialsRefreshToken(context.subscription, context.registry, context);
```

Comment:
After I read the usage, I think you should provide two APIs. 

1. acquireRegistryAccessToken: it's to get the access token to access registry runtime api
2. acquireRegistryLoginCredential: it's to get the docker login credential, basically username and password

Your case should call acquireRegistryLoginCredentail.

Your PR probably can take the change for 2). You can create another PR to refactor the code for 1).